index.js

```diff
@@ -113,8 +113,8 @@ function validateLabel(label, processingOption) {
   let error = false;
 
   if (normalize(label) !== label ||
-      (label[3] === "-" && label[4] === "-") ||
-      label[0] === "-" || label[label.length - 1] === "-" ||
+      (checkHyphens && label[3] === "-" && label[4] === "-") ||
```

Comment:
An array is zero based, so a character in the 3 position is `label[2]`, 4 - `label[3]`.
For example, when `checkHyphens = true` the result must be:
`ab--x` - invalid
`abc--x` - valid